mesonbuild/interpreter/interpreter.py

```diff
@@ -1427,7 +1424,7 @@ def notfound_program(self, args):
     # TODO update modules to always pass `for_machine`. It is bad-form to assume
     # the host machine.
     def find_program_impl(self, args, for_machine: MachineChoice = MachineChoice.HOST,
-                          required=True, silent=True, wanted='', search_dirs=None,
+                          required=True, silent=True, wanted='', search_dirs=[],
```

Comment:
You should in general *never* pass `[]` (or any other mutable type) as a default parameter in python, since this exact same list gets *reused* for each function call. Not sure who thought that this is a good idea but here we are...

So, instead please keep the `search_dirs=None` and add a `search_dirs = search_dirs or []` at the start of the function.

The following example illustrates the problem:

```py
#!/usr/bin/env python3

def foo(data=[]):
    data += ['foo']
    return data

print(foo())
print(foo())
print(foo())
```

Result:
```
['foo']
['foo', 'foo']
['foo', 'foo', 'foo']
```